### ISSUE TITLE
Breaking errors

### ISSUE THREAD

**Issue by jacwright** [andrewrk/node-s3-client#188]
The graceful-fs causes a crash in node v8.2.1. The module it relies on (which is the one crashing) is `natives` and states in its readme:

> I feel compelled to give you a word of warning if you are considering using this module.
> 
> This module lets you do some creative things with the JavaScript code in Node.js. There are some things here that are basically a recipe for memory leaks, or at the very least, being broken with each new release of Node, since none of these API surfaces are "technically" "supported" by the team managing the Node.js project.

I recommend just using the built-in `fs` module and removing this dependency.

**Comment by WaleedAshraf** [comment]
Yes, this is causing trouble, see here: https://github.com/npm/npm/issues/19063#issuecomment-344190239

Any package which is using graceful-fs<4.x is breaking in node 8.x
Kindly update to latest version or completely remove this dep.

**Comment by dguayrobotiq** [comment]
What's up with this issue? I'm getting this error using either node 8.11.3 or 10.13.0:

```
ERROR in ./node_modules/natives/index.js
Module not found: Error: Can't resolve 'module' in '/Users/<mydirectory>/node_modules/natives'
 @ ./node_modules/natives/index.js 2:13-30
 @ ./node_modules/s3/node_modules/graceful-fs/fs.js
 @ ./node_modules/s3/node_modules/graceful-fs/graceful-fs.js
 @ ./node_modules/s3/lib/index.js
 @ ./src/services/Daemon.ts
 @ ./src/Index.ts

ERROR in ./node_modules/natives/index.js
Module not found: Error: Can't resolve 'v8' in '/Users/<mydirectory>/node_modules/natives'
 @ ./node_modules/natives/index.js 129:4-17 152:8-21
 @ ./node_modules/s3/node_modules/graceful-fs/fs.js
 @ ./node_modules/s3/node_modules/graceful-fs/graceful-fs.js
 @ ./node_modules/s3/lib/index.js
 @ ./src/services/Daemon.ts
 @ ./src/Index.ts
```
